src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -536,7 +536,7 @@ template <class PType>
 struct typed_program_node : public typed_program_node_base<PType> {
     using typed_program_node_base<PType>::typed_program_node_base;
 
-    program_node& input() const { return program_node::get_dependency(0); }
+    program_node& input(size_t index = 0) const { return program_node::get_dependency(index); }
```

Comment:
why not just use get_dependnecy(index) instead of fixing this?

Reply:
We have the same fix in this PR - https://github.com/openvinotoolkit/openvino/pull/13576. I just already made those changes for this operation.